README.md

```diff
@@ -1,32 +1,576 @@
+- [Installation Instructions](#installation-instructions)
+  - [In-App Messaging](#in-app-messaging)
+    - [In-App Messaging Deeplinks](#in-app-messaging-deeplinks)
+  - [Push Notifications](#push-notifications)
+    - [If you already use Push Notifications](#if-you-already-use-push-notifications)
+    - [Creating and Inputting the Android Server Key](#creating-and-inputting-the-android-server-key)
+    - [Advanced Configuration](#advanced-configuration)
+  - [Sending Events](#sending-events)
+    - [Sending Named Events](#sending-named-events)
+    - [Event Payloads](#event-payloads)
+    - [Send User Properties](#send-user-properties)
+    - [Sending Virtual Economy Events](#sending-virtual-economy-events)
+    - [Sending IAP Events and IAP Validation](#sending-iap-events-and-iap-validation)
+    - [IAP Functions](#iap-functions)
+    - [Enabling IAP Receipt Validation](#enabling-iap-receipt-validation)
+    - [Verifying IAP Receipt Validation](#verifying-iap-receipt-validation)
+  - [Integrating Resource A/B Testing](#integrating-resource-ab-testing)
+  - [Testing your integration](#testing-your-integration)
+- [Upgrade Instructions](#upgrade-instructions)
+- [How to run the demo](#how-to-run-the-demo)
+- [How to build the SDK](#how-to-build-the-sdk)
+- [Contributing](#contributing)
+- [License](#license)
 
 Getting started
----------------
+=
 Have a look at the quick integration guide at http://docs.swrve.com/developer-documentation/37926233/
 
 Requirements
-------------
+-
+
 ### Android 2.3.3+
 The SDK supports Android 2.3.3+ but will handle older versions with a dummy SDK.
 
 ### Gradle (distributed with the SDK)
 Used to build the SDK and its dependencies.
 
+Installation Instructions
+=
+
+In-App Messaging
+-
+Integrate the in-app messaging functionality so you can use Swrve to send personalized messages to your app users while they’re using your app. If you’d like to find out more about in-app messaging, see [Intro to In-App Messages](http://docs.swrve.com/user-documentation/in-app-messaging/intro-to-in-app-messages/).
+
+Before you can test the in-app message feature in your game, you need to create an In App Message campaign in the Swrve Dashboard.
+
+### In-App Messaging Deeplinks
+
+When creating in-app messages in Swrve, you can configure message buttons to direct users to perform a custom action when clicked. For example, you might configure a button to direct the app user straight to your app store. To enable this feature, you must configure deeplinks by performing the actions outlined below. For more information about creating in-app messages in Swrve, see [Creating In-App Messages](http://docs.swrve.com/user-documentation/in-app-messaging/creating-in-app-messages/).
+
+Swrve's default deeplink behaviour is to treat custom actions as URLs and therefore use your existing custom URL scheme. Before handling deeplinks in Swrve, you’ll need to register a custom URL scheme in your `AndroidManifest.xml`.
```

Comment:
I think for this you should mention that it is possible to handle deeplinks without registering a custom url scheme in the manifest, but if you want to deeplink in from outside the app then it must be mentioned in the manifest.